setup.cfg

```diff
@@ -43,15 +43,15 @@ console_scripts =
 [options.extras_require]
 doc =
     furo
-    sphinx
-    sphinx-autodoc-typehints
-    sphinxcontrib-autoprogram
+    sphinx~=3.0
+    sphinx-autodoc-typehints~=1.10
+    sphinxcontrib-autoprogram~=0.1.0
 test =
-    filelock
-    pytest
-    pytest-cov
-    pytest-mock
-    pytest-xdist>=1.34
+    filelock~=3.0
+    pytest>=5,<=6
```

Comment:
For python 2 the latest version is 2, for python 3 6. Any reason we're not pulling in the latest for all above 4?